examples/hf_serialization.py

```diff
@@ -60,23 +110,40 @@ def serialize_model(
             is purely optional, and it allows for multiple models to be
             serialized to the same directory. A good example are Stable
             Diffusion models. Default is "model".
+        force: Force upload serialized tensors to `output_prefix`
+            even if they already exist
     """
 
-    os.makedirs(model_directory, exist_ok=True)
-    dir_prefix = f"{model_directory}/{model_prefix}"
+    if model_prefix is None:
+        model_prefix = "model"
 
+    dir_prefix = f"{model_directory}/{model_prefix}"
+    config_file_exists, weights_file_exists = (
+        check_file_exists(f"{dir_prefix}-config.json"),
+        check_file_exists(f"{dir_prefix}.tensors"),
+    )
     if config is None:
         config = model
     if config is not None:
-        if hasattr(config, "to_json_file"):
-            config.to_json_file(f"{dir_prefix}-config.json")
```

Comment:
Why remove the propagation of the configuration file?

Reply:
The configuration file is still intended on being propagated. It's just now being converted to a `dict` and dumped as a JSON string. The only difference with this implementation is that it no longer tries to retrieve the `.to_json_file` method, as I was having trouble writing the file to S3 that way. In this case, it uses the `.to_dict` method instead and retains the case handling of checking directly if `config` is a `dict`.

Reply:
It's still saved, but it writes to a stream opened by `open_stream()` rather than using `.to_json_file()`, which presumably doesn't support object storage.